crypto/types.go

```diff
@@ -124,6 +123,6 @@ func (e InvalidInputsError) Error() string {
 
 // IsInvalidInputsError checks if the input error is of a InvalidInputsError type
 func IsInvalidInputsError(err error) bool {
-	var errInvalidInputsError InvalidInputsError
-	return errors.As(err, &errInvalidInputsError)
+	_, ok := err.(*InvalidInputsError)
```

Comment:
This prevents wrapped errors. For example a common case is `InvalidInputsError` wrapped into another higher level error level, and still in that case we want `IsInvalidInputsError` to return true. We mostly use Is method on higher level errors for this goal.

Reply:
You're right. I changed this for 2 reasons though:
 - there are no wrapped errors in this case.
 - `errors.As` as implemented previously wasn't giving the correct behaviour, I'm happy to use it back if there is a fix.